### PR TITLE
CalculateAverage_gonix update

### PR DESCRIPTION
**Pull request by gonix**
Minor updates here and there, shaves off ~5% of execution time on my machine.

#### Check List:
- [x] Tests pass (`./test.sh <username>` shows no differences between expected and actual outputs)
- [x] All formatting changes by the build are committed
- [x] Your launch script is named `calculate_average_<username>.sh` (make sure to match casing of your GH user name) and is executable
- [x] Output matches that of `calculate_average_baseline.sh`
- [x] For new entries, or after substantial changes: When implementing custom hash structures, please point to where you deal with hash collisions (line number)
  - See lines 208 and 239

* Execution time: ~1.75s
* Execution time of reference implementation: ~145s
* Machine: Intel i7-11800H @ 2.30GHz (max 4.6GHz), 8 cores (16 threads), 32GB RAM

### CODE REVIEW

File: src/main/java/dev/morling/onebrc/CalculateAverage_gonix.java [hunk 1]
```diff
@@ -205,27 +205,15 @@ public Stream<Entry> stream() {
 
     private void add(ByteBuffer buf, int start, long tailAndLen, int hash, int measurement) {
         int idx = hash & INDEX_MASK;
-        while (true) {
-            if (index[idx] != 0) {
-                int offset = index[idx];
-                if (keyEqual(offset, buf, start, tailAndLen)) {
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/onebrc/CalculateAverage_gonix.java:211]: How do you deal with key collisions now?

Comment (gonix) [reply src/main/java/dev/morling/onebrc/CalculateAverage_gonix.java:211]: The same as before: by comparing the key. I've moved entry update logic into the `keyEqual()` method and renamed it to `update()`, which returns `false` if the keys are not equal (collision), in which case the search continues to the next slot.

Comment (gunnarmorling) [reply src/main/java/dev/morling/onebrc/CalculateAverage_gonix.java:211]: Ah yes, got it. Thx!

**Comment by gunnarmorling** [comment]
00:03.824 now.

**Pull request merged**